### PR TITLE
ARROW-14839 [R] test-fedora-r-clang-sanitizer job failing due to snappy causing a sanitizer error

### PR DESCRIPTION
**Pull request by jonkeane**

### CODE REVIEW

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-fedora-r-clang-sanitizer

File: cpp/thirdparty/versions.txt [hunk 1]
```diff
@@ -123,7 +123,7 @@ DEPENDENCIES=(
   "ARROW_PROTOBUF_URL protobuf-${ARROW_PROTOBUF_BUILD_VERSION}.tar.gz https://github.com/google/protobuf/releases/download/${ARROW_PROTOBUF_BUILD_VERSION}/protobuf-all-${ARROW_PROTOBUF_BUILD_VERSION:1}.tar.gz"
   "ARROW_RAPIDJSON_URL rapidjson-${ARROW_RAPIDJSON_BUILD_VERSION}.tar.gz https://github.com/miloyip/rapidjson/archive/${ARROW_RAPIDJSON_BUILD_VERSION}.tar.gz"
   "ARROW_RE2_URL re2-${ARROW_RE2_BUILD_VERSION}.tar.gz https://github.com/google/re2/archive/${ARROW_RE2_BUILD_VERSION}.tar.gz"
-  "ARROW_SNAPPY_URL snappy-${ARROW_SNAPPY_BUILD_VERSION}.tar.gz https://github.com/google/snappy/archive/${ARROW_SNAPPY_BUILD_VERSION}.tar.gz"
+  "ARROW_SNAPPY_URL snappy-${ARROW_SNAPPY_BUILD_VERSION}.tar.gz https://codeload.github.com/google/snappy/tar.gz/master"
```
[the remark this answers is not in the document]
Comment (jonkeane) [reply cpp/thirdparty/versions.txt:126]: Yes, if I can get this to work, that is what we'll need to do (or roll our own snappy "release" that we host)

**Comment by pitrou** [comment]
@github-actions crossbow submit test-fedora-r-clang-sanitizer

**Comment by jonkeane** [comment]
Thank you @pitrou ! I've undone some of the stuff I did in my testing, and now we've got passing builds. I'll run a full r group run to confirm this isn't doing something else funny on any of the more exotic architectures and then merge.

**Comment by jonkeane** [comment]
@github-actions crossbow submit -g r

**Comment by pitrou** [comment]
Uh, we should wait for the Snappy change to be merged before updating this. I don't think we want to fetch Snappy from some random third-party fork.

**Comment by jonkeane** [comment]
The following failures in the group are all explained elsewhere:

homebrew-r-autobrew — failing on arrow@HEAD / in the nightlies, https://issues.apache.org/jira/browse/ARROW-14802
test-r-gcc-11  —  resolved in #11795
test-r-linux-as-cran —  resolved in #11795

**Comment by jonkeane** [comment]
Ok, I've got an alternative that patches snappy 1.1.9 instead of using a special self-hosted version of snappy in #11875 I'm happy to do either that PR or this one (well, after we find a way to host our custom patched snappy source). But _would_ like to merge one or the other so we have some clean clang sanitizer CI in the weeks leading up to the release burn down.

**Comment by pitrou** [comment]
I'm ok with the patch solution.

**Pull request closed without merging**